### PR TITLE
store/tikv: check `Deadlock` in `KeyError`

### PR DESCRIPTION
**Pull request by coocood**
### What problem does this PR solve? 
Adapt for new kvproto.

### What is changed and how it works?
- set `forUpdateTS` in `PrewriteRequest`
- check `Deadlock` in `KeyError`.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by coocood** [comment]
@youjiali1995 @tiancaiamao @jackysp PTAL

**Comment by jackysp** [comment]
@youjiali1995 TiDB only needs one approved to merge the pr.

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -616,6 +617,9 @@ func (c *twoPhaseCommitter) pessimisticLockSingleBatch(bo *Backoffer, batch batc
 				}
 				return errors.Trace(conditionPair.Err())
 			}
+			if deadlock := keyErr.Deadlock; deadlock != nil {
+				return errors.New("deadlock")
```
Comment (jackysp) [inline store/tikv/2pc.go]: It's better to use the same error as MySQL.

Comment (coocood) [reply store/tikv/2pc.go]: It's replaced to MySQL deadlock error here:
https://github.com/pingcap/tidb/blob/master/session/tidb.go#L166

Comment (ngaut) [reply store/tikv/2pc.go]: This is a bad design while matching string. Could you refactor it?

**Pull request merged**